util/io/src/lib.rs

```diff
@@ -123,58 +152,120 @@ pub trait IoHandler<Message>: Send + Sync where Message: Send + Sync + Clone + '
 	/// Called when a broadcasted message is received. The message can only be sent from a different IO handler.
 	fn message(&self, _io: &IoContext<Message>, _message: &Message) {}
 	/// Called when an IO stream gets closed
+	#[cfg(feature = "mio")]
 	fn stream_hup(&self, _io: &IoContext<Message>, _stream: StreamToken) {}
 	/// Called when an IO stream can be read from
+	#[cfg(feature = "mio")]
 	fn stream_readable(&self, _io: &IoContext<Message>, _stream: StreamToken) {}
 	/// Called when an IO stream can be written to
+	#[cfg(feature = "mio")]
 	fn stream_writable(&self, _io: &IoContext<Message>, _stream: StreamToken) {}
 	/// Register a new stream with the event loop
+	#[cfg(feature = "mio")]
 	fn register_stream(&self, _stream: StreamToken, _reg: Token, _event_loop: &mut EventLoop<IoManager<Message>>) {}
 	/// Re-register a stream with the event loop
+	#[cfg(feature = "mio")]
 	fn update_stream(&self, _stream: StreamToken, _reg: Token, _event_loop: &mut EventLoop<IoManager<Message>>) {}
 	/// Deregister a stream. Called whenstream is removed from event loop
+	#[cfg(feature = "mio")]
 	fn deregister_stream(&self, _stream: StreamToken, _event_loop: &mut EventLoop<IoManager<Message>>) {}
 }
 
```

Comment:
These `annotations` are not that nice, can we get rid off them by splitting the `IoHandler trait` into two separate traits in each module (service_mio and service_non_mio)?

Reply:
Sounds good to me, one trait should probably require the other and in `non-mio` implementation it can be just an empty marker trait.